### PR TITLE
chore: documentation cleanup for UI shell architecture

### PR DESCRIPTION
**Pull request by joshsmithxrm**
## Summary

Major documentation cleanup to align with the new UI shell architecture pattern. The extension is transitioning from a complex Clean Architecture TypeScript application to a simpler UI shell that calls CLI commands.

### Key Changes

- **Rewrote CLAUDE.md** (~500 → ~180 lines) - Simplified for UI shell pattern, removed Clean Architecture rules
- **Simplified .claude/ structure** - Reduced from 18 commands to 3 essential ones
- **Deleted obsolete architecture docs** - Removed 27+ files including Clean Architecture guides, testing guides, domain patterns
- **Renumbered ADRs** - Cleaned up after deleting obsolete ADRs (0002, 0003)
- **Fixed all broken references** - Updated 14 files with broken links to deleted docs
- **Migrated project tracking** - Moved from markdown files to GitHub Issues

### Files Deleted

- `docs/architecture/CLEAN_ARCHITECTURE_*.md` (3 files)
- `docs/architecture/*_PATTERNS.md` (4 files - VALUE_OBJECT, DOMAIN_SERVICE, MAPPER, REPOSITORY)
- `docs/architecture/CODE_QUALITY_GUIDE.md`, `LOGGING_GUIDE.md`, `PANEL_ARCHITECTURE.md`
- `docs/testing/*.md` (3 files)
- `docs/future/*`, `docs/quality/*`, `docs/requirements/*`, `docs/work/*`
- `docs/adr/0002_*.md`, `docs/adr/0003_*.md`
- `.claude/TROUBLESHOOTING.md`, obsolete command files

### Files Updated

- `CLAUDE.md` - Complete rewrite for UI shell pattern
- `CONTRIBUTING.md` - Simplified testing section
- `.claude/commands/*.md` - Simplified command files
- `.claude/templates/*.md` - Simplified panel guides
- `docs/README.md` - Updated index
- `docs/architecture/*.md` - Removed broken references

## Test Plan

- [x] `npm run compile` passes
- [x] No broken documentation links remain
- [x] ADR numbering is sequential (0001-0005)

### CODE REVIEW

**Review by joshsmithxrm — COMMENTED** [review]
Fixed in 2827b84. Removed the broken reference to ODATA_DOMAIN_PATTERN.md (which was deleted). Kept the valid reference to STATIC_FACTORY_PATTERN.md which still exists.

File: docs/adr/0001_ESLINT_SUPPRESSIONS.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+| `@typescript-eslint/no-explicit-any` | 8 | Test validation of edge cases |
+| `complexity` | 4 | Parameter assignment, not logic complexity |
+| `local-rules/no-presentation-methods-in-domain` | 3 | OData query building (design decision) |
+| `prefer-const` | 2 | MSAL library mutation requirements |
+| Other | 4 | Various justified cases |
+
+## References
+
+- [ODATA_DOMAIN_PATTERN.md](../architecture/ODATA_DOMAIN_PATTERN.md) - OData in domain rationale
+- [STATIC_FACTORY_PATTERN.md](../architecture/STATIC_FACTORY_PATTERN.md) - Static factory pattern
```
[the remark this answers is not in the document]
Comment (joshsmithxrm) [reply docs/adr/0001_ESLINT_SUPPRESSIONS.md]: Fixed in 2827b84. Removed the broken reference to ODATA_DOMAIN_PATTERN.md (which was deleted). Kept the valid reference to STATIC_FACTORY_PATTERN.md which still exists.
